tests/ha/test_multi_server_pod_failure.py

```diff
@@ -629,3 +629,584 @@ def test_writes_till_kserver_pods_fail(self):
+            pod_list.remove(pod_name)
+        assert_utils.assert_equal(counter, 0, "Services on some server pods not stopped.")
+        LOGGER.info("Step 5: Services of server pods are in offline state")
+
+        LOGGER.info("Step 6: Check services status on remaining pods %s", server_data_pod_list)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=server_data_pod_list,
+                                                           fail=False)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 6: Services of remaining pods are in online state")
+
+        LOGGER.info("Step 7: Verify status for In-flight DELETEs while %s (K) server pods were"
+                    "going down", self.kvalue)
+        LOGGER.info("Waiting for background DELETEs thread to join. Waiting for %s seconds to "
+                    "collect the queue logs", HA_CFG["common_params"]["60sec_delay"])
+        thread.join()
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "Background process failed to do deletes")
+        event_del_bkt = del_resp[0]
+        fail_del_bkt = del_resp[1]
+        rem_bkts_aftr_del = s3_test_obj.bucket_list()[1]
+        assert_utils.assert_false(len(fail_del_bkt),
+                                  f"Bucket deletion failed when cluster was online {fail_del_bkt}")
+        assert_utils.assert_true(len(rem_bkts_aftr_del) < del_bucket,
+                                 "Some bucket deletion expected during pods going down")
+        LOGGER.info("Step 7: Verified status for In-flight DELETEs while %s (K) server pods were"
+                    "going down", self.kvalue)
+
+        LOGGER.info("Step 8: Perform DELETEs on remaining FailedToDelete buckets "
+                    "when server pods were going down.")
+        fail_del_op = Queue()
+        args = {'test_prefix': self.test_prefix, 'test_dir_path': self.test_dir_path,
+                'skipput': True, 'skipget': True, 'bkt_list': event_del_bkt,
+                'output': fail_del_op}
+        self.ha_obj.put_get_delete(event, s3_test_obj, **args)
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = fail_del_op.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        event_del_bkt = del_resp[0]
+        fail_del_bkt = del_resp[1]
+        assert_utils.assert_false(len(event_del_bkt) or len(fail_del_bkt),
+                                  f"Failed to delete buckets: either {event_del_bkt} or"
+                                  f" {fail_del_bkt}")
+        LOGGER.info("Step 8: Successfully performed DELETEs on remaining FailedToDelete buckets "
+                    "when server pods were going down")
+
+        LOGGER.info("Step 9: Verify read on the remaining %s buckets.", len(rem_bkts_aftr_del))
```

Comment:
rem_bkts_aftr_del will also include buckets which we deleted in step 8 right?
 these are failed to delete bucket when server pod was going down. This should not be part of this as step 8 is already deleting the same

Reply:
This is to read from the remaining buckets.